galasa-parent/dev.galasa.framework.api.authentication/src/main/java/dev/galasa/framework/api/authentication/internal/routes/AuthTokensRoute.java

```diff
@@ -76,36 +76,76 @@ public AuthTokensRoute(
     /**
      * GET requests to /auth/tokens return all the tokens stored in the tokens
      * database, sorted by creation date order by default.
+     * This endpoint takes an optional query paramater 'loginId' for e.g loginId=admin
+     * Passing it returns a filtered list of token records stored in the auth store that matches the given login ID
      */
     @Override
     public HttpServletResponse handleGetRequest(String pathInfo, QueryParameters queryParams,
             HttpServletRequest request, HttpServletResponse response)
             throws ServletException, IOException, FrameworkException {
 
         logger.info("handleGetRequest() entered");
+        
+        List<AuthToken> tokensToReturn = new ArrayList<>();
+        String loginId = queryParams.getSingleString(AuthenticationServlet.QUERY_PARAM_LOGIN_ID, null);
+        
+        if(queryParams.checkAtLeastOneQueryParameterPresent("loginId")){
+            validateLoginId(loginId, pathInfo);        
+        }
+
+        if(loginId == null){
+            tokensToReturn = handleGetAllTokens();
+        }
+        else{
+            tokensToReturn = handleGetTokensByLoginId(loginId);
+        }
+
+        return getResponseBuilder().buildResponse(request, response, "application/json",
+                getTokensAsJsonString(tokensToReturn), HttpServletResponse.SC_OK);
+    }
+
+    public List<AuthToken> handleGetAllTokens() throws ServletException, IOException, FrameworkException{
```

Comment:
This can be a private method called getAllTokens() - we don't need the "handle" bit